JH6/src/main/java/homework/transport/Bicycle.java

```diff
@@ -0,0 +1,92 @@
+package homework.transport;
+
+/**
+ * Class contains data of travelling by bicycle.
+ */
+public class Bicycle implements Moveable {
+
+    /**
+     * Average speed moving by bicycle(km per hour)
+     */
+    private double averageSpeed = 0.0;
+
+    /**
+     * Name of the way of travelling
+     */
+    public static final String NAME = "Bicycle";
+
+    /**
+     * Current travel time (hours)
+     */
+    private double travelTime = 0.0;
+
+    /**
+     * Object, that calculate distance
+     */
+    private Router router = new Router();
+
+    /**
+     * Constructor creates the vehicle, set average speed
+     *
+     * @param averageSpeed average speed of travelling afoot
+     */
+    public Bicycle(double averageSpeed) throws Exception {
+        if (averageSpeed > 0) {
+            this.averageSpeed = averageSpeed;
+        } else {
+            throw new Exception("Incorrect speed \"" + averageSpeed + "\"");
+        }
+    }
+
+    /**
+     * Method sets start coordinates of travelling
+     *
+     * @param x coordinate x
+     * @param y coordinate y
+     */
+    @Override
+    public void setStartCoordinates(double x, double y) {
+        router.setStartCoordinates(x, y);
+    }
+
+    /**
+     * Method emulates moving to the next point, which coordinates got as
+     * parameters. It calculates current travel time
+     *
+     * @param x coordinate x
+     * @param y coordinate y
+     */
+    @Override
+    public void moveToNextCheckpoint(double x, double y) {
+        travelTime += router.getDistance(x, y) / averageSpeed;
+    }
+
+    /**
+     * Method returns current travel time in special format
```

Comment:
It's better to return in some generic format and leave special formatting to caller